src/services/dashboards/modules/dashboard-toolset/DashboardDateDropdown.vue

```diff
@@ -128,6 +134,51 @@ export default defineComponent({
             state.customRangeModalVisible = false;
         };
 
+        const setInitialDateRange = () => {
+            const _start = props.dateRange?.start ?? '';
+            const _end = props.dateRange?.end ?? '';
+            const start = {
+                y: _start.split('-')[0],
+                m: _start.split('-')[1],
+                d: _start.split('-')[2],
+            };
+            const end = {
+                y: _end.split('-')[0],
+                m: _end.split('-')[1],
+                d: _end.split('-')[2],
+            };
+            const current = {
+                y: dayjs.utc().format('YYYY'),
+                m: dayjs.utc().format('MM'),
+                d: dayjs.utc().format('DD'),
+            };
+
+            // 1. default month => start is (month 'current' + day '1'), end is (month 'current + day 'today')
+            // Index 0 is 'Current' menu index
+            if (!start.d || !end.d
+            || (_start === `${current.y}-${current.m}-01`
+            && _end === `${current.y}-${current.m}-${current.d}`)
+            ) {
+                return 0;
+            }
```

Comment:
suggestion: I think this is safer
```typescript
const _current = dayjs.utc();
const _start = dayjs.utc(props.dateRange?.start);
const _end = dayjs.utc(props.dateRange?.end);
if (!props.dateRange?.start || !props.dateRange?.end || (_start.isSame(_current.startOf('month'), 'day') && _end.isSame(_current.endOf('month'), 'day'))) { return 0; }
```

Reply:
`isSame` 
Passing in `month` will check `month` and `year`. Passing in `day` will check `day`, `month`, and `year`.